generator/nanopb_generator.py

```diff
@@ -59,9 +59,15 @@
     ''' + '\n')
     raise
 except (ValueError, SystemError, ImportError):
-    # Probably invoked directly instead of via installed scripts.
-    import proto.nanopb_pb2 as nanopb_pb2
-    from proto._utils import invoke_protoc
+    try:
+        # Probably invoked directly instead of via installed scripts.
+        import proto.nanopb_pb2 as nanopb_pb2
+        from proto._utils import invoke_protoc
+    except ImportError:
+        # Invoked using bazel
```

Comment:
This starts to feel like a hack on top of hack to handle the import paths in different situations. But I have no better alternative to offer, my understanding of Python import mechanism is a bit lacking.

Reply:
This is absolutely a hack, and to be honest, this was the biggest headache in getting this to work. It is bothering me so it's certainly something that I would like to improve upon in general. I will have a bit more of a look around at other projects. I feel that it should be possible to find a common method of importing that works with pip, bazel and checked out repos.

Reply:
I'm not 100% on this but I get the sense that bazel has some gaps in python support for the various techniques python uses to resolve imports, especially those that don't always mirror the paths on the file system. This bazel issue I filed comes to mind here https://github.com/bazelbuild/bazel/issues/11414